gnomad_hail/resources/grch38/reference_data.py

```diff
@@ -13,6 +13,9 @@
 
 def _import_purcell_5k(path) -> hl.Table:
     p5k = hl.import_locus_intervals(path, reference_genome='GRCh37')
+    rg37 = hl.get_reference('GRCh37')
+    rg38 = hl.get_reference('GRCh38')
+    rg37.add_liftover('gs://hail-common/references/grch37_to_grch38.over.chain.gz', rg38)
```

Comment:
[Hail docs](https://hail.is/docs/0.2/genetics/hail.genetics.ReferenceGenome.html#hail.genetics.ReferenceGenome.add_liftover) say "This method can only be run once per reference genome." Will this break if we need to use liftover in another import? Maybe this should be extracted into a function like `ensure_liftover` that first checks if the chain file has been registered using `has_liftover`.

Reply:
Or is there a way to copy the reference genome objects, add the liftover to them, and use the copies in the import. That way this wouldn't modify the global reference genomes.

Reply:
From what I can tell reading the docs, we can't copy them.  I can make an `ensure_liftover` resource util unless someone knows how to copy the objects.

Reply:
It looks like even if a copy was possible, it would also be registered globally as well.

> Note that constructing a new reference genome, either by using the class constructor or by using ReferenceGenome.read() will add the reference genome to the list of known references; it is possible to access the reference genome using get_reference() anytime afterwards.
> 
> -- https://hail.is/docs/0.2/genetics/hail.genetics.ReferenceGenome.html

Reply:
Good catch! should we just do 
```
if not rg37.has_liftover:
    rg37.add_liftover('gs://hail-common/references/grch37_to_grch38.over.chain.gz', rg38)
```

I feel like adding a function for this is overkill..

Reply:
That function may already exist in liftover.py:
https://github.com/macarthur-lab/gnomad_hail/blob/ecf5e9cc9b297bfd8a24ca683da62b49e27d71bd/gnomad_hail/utils/liftover.py#L35-L65

cc @ch-kr

Reply:
Always using one function for liftover ensures we're consistent with the chain files. If every function that needs to liftover does `if not has_liftover, add_liftover` and the same chain file isn't used everywhere, then the results of liftover could depend on what order functions are called in.

Reply:
Damn, that function in `liftover.py` isn't safe either :(

Reply:
I think we need our own reference class wrapper if we don't want an error and we want to have a clean `ensure_liftover` mechanism.

Reply:
Otherwise either we let it error out or we assume there is only one possible chain file.

Reply:
Thinking about it, I feel like a simple reference singleton with defaults may well be useful

Reply:
I'm not sure I understand 

> simple reference singleton with defaults

Would this mean adding a new resource class and just returning a reference object with chain files added?

Reply:
well, we could have a default chain file that is stored, and when calling `.add_liftover` it would   make sure that they are the same file. 
All in all, we have quite a few places in the library that deal with reference genomes. I think we should address this more globally outside of this PR. But I think it's worth thinking about a good mechanism to handle them.

Reply:
So I'd vote to make and issue so that we can keep track of this but not alter the PR

Reply:
https://github.com/macarthur-lab/gnomad_hail/issues/162